pandas/io/parsers.py

```diff
@@ -1772,7 +1773,21 @@ def _handle_usecols(self, columns, usecols_key):
                 col_indices = []
                 for u in self.usecols:
                     if isinstance(u, string_types):
-                        col_indices.append(usecols_key.index(u))
+                        if usecols_key.index(u) not in col_indices:
+                            col_indices.append(usecols_key.index(u))
+                        else: # found duplicates in usecols_key.
+                            # get indices of all instances of u in usecols_key. GH11823
+                            u_index = Index(usecols_key).get_loc(u)
+                            # convert the slice or  mask array to index
```

Comment:
I agree. Well, here are some examples of various ``usecols`` with and without duplicates for when ``names`` contain duplicates.
```
pd.read_csv(StringIO("""1,2,3"""), engine='python', header=None, 
            names=['a', 'b', 'a'], usecols=['a'])
Out: 
   a
0  1
pd.read_csv(StringIO("""1,2,3"""), engine='python', header=None, 
            names=['a', 'b', 'a'], usecols=['a','a'])
Out: 
   a  a
0  3  3
pd.read_csv(StringIO("""1,2,3"""), engine='python', header=None, 
            names=['a', 'b', 'a'], usecols=['a','b'])
Out: 
   a  b
0  1  2
pd.read_csv(StringIO("""1,2,3"""), engine='python', header=None, 
            names=['a', 'b', 'a'], usecols=['a','b','a'])
Out: 
   a  b  a
0  3  2  3
```
In retrospect I'm not sure if this complexity is worth it here, and in any case one can still argue that the above behavior is not ideal ...

What about throwing an error when duplicated columns exist with ``usecols``? Or leave the python engine alone and edit only the c engine parser so that the "(b) looks like a bug" part of the original issue #11823 is addressed, but not the "(a) different" part?